### PR TITLE
Add support for copying files

### PR DESCRIPTION
**Pull request by znewman01**
Hi Jon! Not sure if you know me, I sit a floor down from you (err...not lately,
but you you know what I mean). Thanks for the library, it's been useful.

I've added, in the style of `scp(1)`, support for uploading/downloading files:

```rust
MachineSetup::new("c5.xlarge", "ami-e18aa89b", |ssh| {
    ssh.upload(
        Path::new("/path/on/local/machine.rar"),
        Path::new("dest/on/remote/machine.rar")
    )?;
    ssh.download(
        Path::new("remote-machine/business.xls"),
        Path::new("/tmp/business.xls")
    )?;
    Ok(())
});
```

I've verified that it works in a quick example, and I'm planning to use it for
some experiments very soon (copying a source tarball up, then building it, then
copying the binaries back down). I'll let you know if that doesn't go well.

I'm almost tempted to implement this as `Session.open(&Path) -> Result<ssh2::File, _>` to enable:

``` rust
let remote_src = ssh.open(Path::new("file"))?;
let remote_dst = ssh.open(Path::new("file"))?;
std::io:copy(remote_src, remote_dst)?;
```

but this oughta suffice for most use cases.

### CODE REVIEW

File: src/ssh.rs [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 use ssh2;
+use std::fs::File;
+use std::io;
 use std::net::{SocketAddr, TcpStream};
 use std::path::Path;
 use std::thread;
 use std::time::{Duration, Instant};
 
+#[derive(Debug, Fail)]
+enum SshError {
```
Comment (jonhoo) [inline src/ssh.rs]: This can probably be a `struct FileTransferError` rather than an `enum`, since there aren't any other variants :)

**Review by jonhoo — APPROVED** [review]
This looks great modulo the error change I pointed out above!

File: src/ssh.rs (hunk 1 above)
Comment (znewman01) [reply src/ssh.rs]: Done, fair :)

**Comment by znewman01** [comment]
@jonhoo ready for you!

**Comment by jonhoo** [comment]
Beautiful, thank you! I'm in a weird position where I want to wait for #23 before doing another (breaking) release, but I don't want to release it yet because I haven't had a chance to test it properly yet. Are you running on top of that code, and if so, can you also confirm that it works?

**Comment by znewman01** [comment]
I haven't been using #23 (don't need it for my use case), though I suppose it'd be pretty easy for me to test it out (though I wouldn't be exercising the Azure code or even multiple regions within AWS). Why don't I take a crack this weekend and see if it's not too much work?

**Comment by jonhoo** [comment]
That'd be awesome! Even just getting more confidence that the updated AWS code works, and that the API is okay to use would be super helpful (cc @akshayknarayan).

**Comment by akshayknarayan** [comment]
btw, #23 moves the ssh functionality to a separate crate - https://github.com/akshayknarayan/sessh. It has similar functionality to what you want (https://docs.rs/sessh/0.1.3/sessh/struct.Session.html#method.scp_recv), but your convenience methods would be useful there as well!

**Comment by znewman01** [comment]
oh cool, maybe I'll send a (mostly copy+paste) PR over there

**Pull request merged**